core/che-core-db/src/main/java/org/eclipse/che/core/db/JNDIDataSourceFactory.java

```diff
@@ -0,0 +1,55 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2017 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.core.db;
+
+
+import org.apache.tomcat.dbcp.dbcp2.BasicDataSource;
+import org.apache.tomcat.dbcp.dbcp2.BasicDataSourceFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.naming.Context;
+import javax.naming.Name;
+import javax.naming.spi.ObjectFactory;
+import java.util.Hashtable;
+import java.util.Properties;
+
+public class JNDIDataSourceFactory implements ObjectFactory {
```

Comment:
I understood why you're using JNDI resources.
but for example https://tomcat.apache.org/tomcat-8.0-doc/api/org/apache/tomcat/dbcp/dbcp2/BasicDataSourceFactory.html is already a JNDI factory. So you could directly use that JNDI factory as well.

Also do you have considered https://github.com/brettwooldridge/HikariCP#initialization
https://brettwooldridge.github.io/HikariCP/ because dbcp2 has not the best performances vs c3p0 or hikari ?